dkg/nodesigs.go

```diff
@@ -50,13 +51,17 @@ func newNodeSigBcast(operatorAmt int, bcastComp *bcast.Component) nodeSigBcast {
 }
 
 // broadcastCallback is the default bcast.Callback for nodeSigBcast.
-func (n *nodeSigBcast) broadcastCallback(_ context.Context, _ peer.ID, _ string, msg proto.Message) error {
+func (n *nodeSigBcast) broadcastCallback(ctx context.Context, _ peer.ID, _ string, msg proto.Message) error {
 	response, ok := msg.(*dkgpb.MsgNodeSig)
 	if !ok {
 		return errors.New("invalid node sig type")
 	}
 
-	n.otherSigs <- response
+	select {
```

Comment:
lets do verification here, as soon as possible. Since your logic below can be broken if byzantine peers spam invalid data. Also as suggested, store in the map here after verification.